### PR TITLE
cpu: fix building for js

### PR DESCRIPTION
**Pull request by paralin**
Avoid importing the x/sys/cpu package for the js build tag.

Mark that CPU target has having none of the CPU features.

Fixes the build against gopherjs.

### CODE REVIEW

**Comment by marten-seemann** [comment]
1. What's the use case for this?
2. Why are you exposing these as variables, why not unexported constants?
3. We'd also need to make the equivalent changes to https://github.com/marten-seemann/qtls-go1-16 and https://github.com/marten-seemann/qtls-go1-18, unfortunately.

File: cpu.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import (
+	"golang.org/x/sys/cpu"
+)
+
+var (
+	CPUHasAES       = cpu.X86.HasAES
+	CPUHasAESCBC    = cpu.S390X.HasAESCBC
+	CPUHasAESCTR    = cpu.S390X.HasAESCTR
+	CPUHasAESGCM    = cpu.S390X.HasAESGCM
```
Comment (egonelbre) [inline cpu.go]: Why is this using `cpu.S390X`?

File: cpu.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+// +build !js
+
+package qtls
+
+import (
+	"golang.org/x/sys/cpu"
+)
+
+var (
+	CPUHasAES       = cpu.X86.HasAES
```
Comment (egonelbre) [inline cpu.go]: Why is this using `cpu.X86`?

File: cpu.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	CPUHasAESCBC    = cpu.S390X.HasAESCBC
+	CPUHasAESCTR    = cpu.S390X.HasAESCTR
+	CPUHasAESGCM    = cpu.S390X.HasAESGCM
+	CPUHasPCLMULQDQ = cpu.X86.HasPCLMULQDQ
+	CPUHasPMULL     = cpu.ARM64.HasPMULL
```
Comment (egonelbre) [inline cpu.go]: Why is this using `cpu.ARM64`?

**Comment by paralin** [comment]
image

1. I run [Quic in the browser](https://github.com/aperturerobotics/bifrost/tree/master/examples/websocket-browser-link) over WebSocket and other underlying transports using either GopherJS or Wasm depending on what support is available.
2. They can be unexported, sure.
3. I'll link updated PRs for 1-16 and 1-17 and will update this one to unexport the variables.

File: cpu.go (hunk 3 above)
Comment (paralin) [reply cpu.go]: @egonelbre The code is the same as was there before - just moved to a different location.

https://github.com/marten-seemann/qtls-go1-17/pull/2/files#diff-aa99d9b5e777a57d11244c217daa11772dae47d3f935d51f06c20fb0557083f9L370

**Comment by paralin** [comment]
Equiv. patches for the other versions:

 - https://github.com/marten-seemann/qtls-go1-15/pull/2
 - https://github.com/marten-seemann/qtls-go1-16/pull/1

Unexported the variables & tested with all 3 versions

**Review by marten-seemann — APPROVED** [review]

**Comment by marten-seemann** [comment]
> https://github.com/marten-seemann/qtls-go1-15/pull/2

We need https://github.com/marten-seemann/qtls-go1-18, not https://github.com/marten-seemann/qtls-go1-15. Sorry for this mess with the qtls versions btw ;)

**Comment by paralin** [comment]
https://github.com/marten-seemann/qtls-go1-18/pull/2

It's a bit of a mess, but I understand why it's necessary.

**Pull request merged**